src/Components/Facilities/Facilities.tsx

```diff
@@ -0,0 +1,310 @@
+import React, { useContext } from "react";
+import { styled } from "@mui/material/styles";
+import Table from "@mui/material/Table";
+import TableBody from "@mui/material/TableBody";
+import TableCell, { tableCellClasses } from "@mui/material/TableCell";
+import TableContainer from "@mui/material/TableContainer";
+import TableHead from "@mui/material/TableHead";
+import TableRow from "@mui/material/TableRow";
+import Paper from "@mui/material/Paper";
+import { useEffect, useState } from "react";
+import axios from "axios";
+import Box from "@mui/material/Box";
+import Modal from "@mui/material/Modal";
+import { RotatingLines } from "react-loader-spinner";
+import noData from "../../assets/noData.png";
+import { Button } from "@mui/material";
+import DeleteForeverTwoToneIcon from "@mui/icons-material/DeleteForeverTwoTone";
+import ModeEditOutlineTwoToneIcon from "@mui/icons-material/ModeEditOutlineTwoTone";
+import AddFacility from "./addFacility";
+import EditFacility from "./editFacility";
+import DeleteFacility from "./deleteFacility";
+import { AuthContext } from "../Context/AuthContext/AuthContext";
+
+export default function Facilities(props) {
+    const [isLoading, setIsLoading] = useState(false);
+    const [id, setId] = useState(null);
+    const [facility, setFacility] = useState(null);
+    const { baseUrl,requstHeaders } = useContext(AuthContext);
+
+
+     //***********************modal consts************* */
+ 
+
+  const style = {
+    position: "absolute" as "absolute",
+    top: "50%",
+    left: "50%",
+    transform: "translate(-50%, -50%)",
+    width: 400,
+    bgcolor: "background.paper",
+    border: "0px solid #ffff",
+    boxShadow: 24,
+    p: 4,
+  };
+
+  const [open, setOpen] = React.useState(false);
+  const handleOpen = () => setOpen(true);
+  const handleClose = () => setModalState("close");
+
+
+  //***********************table consts************* */
+
+  const StyledTableCell = styled(TableCell)(({ theme }) => ({
+    [`&.${tableCellClasses.head}`]: {
+      backgroundColor: theme.palette.common.black,
+      color: theme.palette.common.white,
+    },
+    [`&.${tableCellClasses.body}`]: {
+      fontSize: 14,
+    },
+  }));
+
+  const StyledTableRow = styled(TableRow)(({ theme }) => ({
+    "&:nth-of-type(odd)": {
+      backgroundColor: theme.palette.action.hover,
+    },
+    // hide last border
+    "&:last-child td, &:last-child th": {
+      border: 0,
+    },
+  }));
+
+  const [facilitiesList, setFacilitiesList] = useState([]);
+  
+
+
+   //**********************to get all Facilities************** */
+
+   const getAllFacilities = () => {

```

Comment:
I noticed using this function in multiple components and passing it via props. You can abstract its logic into a custom hook that will handle the data fetching logic in one place. Also when passing a function via props, in every render the components generate a new function with a totally different reference. This makes the children components re-render in every change.